source/fe/mapping_q_generic.cc

```diff
@@ -2589,10 +2589,15 @@ fill_fe_values (const typename Triangulation<dim,spacedim>::cell_iterator &cell,
   data.mapping_support_points = this->compute_mapping_support_points(cell);
   data.cell_of_current_support_points = cell;
 
+  // if the order of the mapping is greater than 1, then do not reuse any cell
+  // similarity information.
+  const CellSimilarity::Similarity computed_cell_similarity =
+    (polynomial_degree == 1 ? cell_similarity : CellSimilarity::none);
```

Comment:
Please also say in the comment why we do this -- namely because the `cell_similarity` passed down to this point was only computed from vertex locations, but that for higher order mappings, that is not enough to tell whether a cell is a translation of another.